dlt/common/typing.py

```diff
@@ -119,7 +120,34 @@ def asstr(self, verbosity: int = 0) -> str:
         ...
 
 
-def extract_type_if_modifier(t: Type[Any]) -> Type[Any]:
+def get_type_name(t: Type[Any]) -> str:
+    """Returns a human-friendly name of type `t`"""
+    if name := getattr(t, "__name__", None):
+        return name  # type: ignore[no-any-return]
+    t = get_origin(t) or t
+    if name := getattr(t, "__name__", None):
+        return name  # type: ignore[no-any-return]
+    return str(t)
+
+
+def is_callable_type(hint: Type[Any]) -> bool:
+    """Checks if `hint` is callable: a function or callable class. This function does not descend
+    into type arguments ie. if Union, Literal or NewType contain callables, those are ignored"""
+    if get_origin(hint) is get_origin(Callable):
+        return True
+    # this skips NewType etc.
+    if getattr(hint, "__module__", None) == "typing":
+        return False
+    if isinstance(hint, FunctionType):
+        return True
+    # this is how we check if __call__ is implemented in the mro
+    if inspect.isclass(hint) and any("__call__" in t_.__dict__ for t_ in inspect.getmro(hint)):
+        return True
+
+    return False
```

Comment:
I think as a last resort we can return the result of `callable(hint)`

Reply:
Also I came up with less convoluted option which works and tests pass
```py
if get_origin(hint) is Callable:
        return True
    # Avoid unnecessary checks for built-in typing types by checking for __call__ attribute
    if callable(hint):
        return True

    # If hint is a class, check if it or any of its parents have an overridden __call__ method
    if inspect.isclass(hint) and any("__call__" in t.__dict__ for t in inspect.getmro(hint)):
        return True

    return False
```

Reply:
I tried many things. and this
```
 if callable(hint):
        return True
```
is true for any class type, at least in python 3.8 because class types are callable
```
Python 3.8.11 (default, Jul  3 2021, 17:53:42)
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class C:
...     pass
...
>>> callable(C)
True
>>>
```
so IMO your code will not work

Reply:
@sultaniman any update how the tests were passing with the code above?